### PR TITLE
actionSynthesis propagate source info

### PR DESCRIPTION
**Pull request by ChrisDodd**
- propagate source info from statements to newly created actions and
  tables; allows later passes to generate errors/warnings referring to
  the code that the tables were created from rather than just a
  meaningless name like tbl_act_14

### CODE REVIEW

File: midend/actionSynthesis.cpp [hunk 1]
```diff
@@ -191,7 +193,7 @@ const IR::Statement* DoSynthesizeActions::createAction(const IR::Statement* toAd
     actions.push_back(action);
     auto actpath = new IR::PathExpression(name);
     auto repl = new IR::MethodCallExpression(actpath);
-    auto result = new IR::MethodCallStatement(repl->srcInfo, repl);
+    auto result = new IR::MethodCallStatement(toAdd->srcInfo, repl);
```
Comment (mbudiu-vmw) [inline midend/actionSynthesis.cpp]: Maybe the expression above should also use toAdd->srcInfo.

**Comment by mbudiu-vmw** [comment]
The tests fail because you are probably using a source position which is after the place where it is used, so resolveReferences refuses to use it. In the generated programs declarations should still come after uses.

**Comment by ChrisDodd** [comment]
Actually, they were failing because they were using a name with no source position (it was compiler generated).  Such names should always be considered after anything with a source location, not before.

File: midend/actionSynthesis.cpp (hunk 1 above)
Comment (ChrisDodd) [reply midend/actionSynthesis.cpp]: makes sense

**Pull request merged**